flake.nix

```
@@ -18,44 +18,47 @@
         config.allowUnfree = true;
         config.allowUnsupportedSystem = true;
       };
-      ctf = pkgs.buildFHSEnv {
+      ctf = pkgs.mkShell {
         name = "ctf-env";
-        targetPkgs = pkgs: let
-          pythonStuff = pkgs.python311Full.withPackages (pp: [
+        buildInputs = let
+          pythonStuff = pkgs.python311.withPackages (pp: [
```

```suggestion
          pythonStuff = pkgs.python312.withPackages (pp: [
```

I think we can safely update the python version